### PR TITLE
hbone listener: allow pods with no service

### PR DESCRIPTION
**Pull request by howardjohn**
TBH not sure this is a great fix, but seems to work and better than NACKing?

### CODE REVIEW

File: pilot/pkg/networking/core/listener_inbound.go [hunk 1]
```diff
@@ -177,6 +177,13 @@ func (lb *ListenerBuilder) buildInboundHBONEListeners() []*listener.Listener {
 		}
 		l.FilterChains = append(l.FilterChains, chains...)
 	}
+	// If there are no filter chains, populate a dummy one that never matches. Envoy doesn't allow no chains, but removing the
```
Comment (stevenctl) [inline pilot/pkg/networking/core/listener_inbound.go:180]: Don't we usually add a "blackhole" chain as the default? Can we still have that as the final on_no_match?

**Review by stevenctl — APPROVED** [review]

File: pilot/pkg/networking/core/listener_inbound.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/listener_inbound.go:180]: Oh good point

Comment (howardjohn) [reply pilot/pkg/networking/core/listener_inbound.go:180]: Updated

**Pull request merged**